### PR TITLE
feat: log version + effective config/env at session startup (redacted)

### LINKED ISSUE #214: feat: log version + effective config/env at session startup to console and log file
[mode: plan]
[model: sonnet]
[effort: medium]

## Summary

At the start of every run (CLI one-shot and serve mode), emit a startup banner that records the build version and the effective, resolved configuration, to both the console and the log file, with secrets redacted.

## Problem / Motivation

Today there is no single, reliable place that says "this is the version running and these are the settings it is running with." Diagnosing a support issue means asking the user which binary they have, then reconstructing the effective config by hand from the TOML file, the `MXLRC_*` environment, and the CLI flags (precedence: CLI > env > file). In Docker deployments there is no on-disk log file by default and the only window is `docker logs`, so whatever startup context is not logged is simply unavailable after the fact.

A self-documenting startup banner makes every session traceable: at a glance you can see what version is running and exactly which settings (and from which source) were in effect, both in `docker logs` and in the rotating log file.

## Proposed Solution

On startup, before the main work begins, emit a banner containing:

- **Build version**: the same embedded `--version` string the binary already reports.
- **Effective configuration**: each setting with its resolved value and, where practical, its source (file / env / CLI / default), so the precedence outcome is visible.
- **Relevant `MXLRC_*` environment**: the env values that participated in resolution.

Output and redaction requirements:

- **Both sinks**: write to the user-facing console (stdout) AND through `slog` so the same diagnostic context lands in the configured log file (and `docker logs`). The console form can be concise/human-readable; the `slog` form should be structured.
- **Redaction**: every secret-bearing field (API tokens, webhook API keys, any credential) must be redacted (e.g. `***`). Reuse the existing config-redaction logic - the same redaction the planned serve-mode Config view (#210) relies on - so there is one source of truth for what counts as a secret.
- Keep it readable: the effective-config dump should be ordered/grouped consistently (mirroring the config sections), not a raw struct print.

Design notes:

- Factor the redaction + effective-config rendering so both this banner and the #210 serve Config view consume it (avoid two divergent redaction lists).
- Consider a verbosity consideration: the full config dump may be noisy at every start - decide whether it is always-on, gated by log level (e.g. full dump at DEBUG, one-line version+key-settings at INFO), or a flag. Capture the decision in the plan.

## Acceptance Criteria

- [ ] On startup of both CLI one-shot runs and serve mode, a banner with the build version is emitted to console and via `slog` to the log file.
- [ ] The effective resolved configuration is included, with per-setting source where practical (file/env/CLI/default).
- [ ] All secret-bearing fields are redacted, proven by an in-package test; redaction reuses the shared logic (same as the #210 Config view).
- [ ] Output reaches both the console and the configured log file / `docker logs`.
- [ ] Log-level / verbosity behavior for the full dump is decided and documented.

## Additional Context

- Token precedence and config resolution live in `internal/config` (CLI > env > file).
- Redaction should be shared with the serve-mode Config view (#210), which also renders effective TOML with secrets redacted.
- Deployment note: the tool commonly runs in Docker with no on-disk log file, so the `slog`/`docker logs` path is the primary capture surface there.

### PR DESCRIPTION
**Pull request by sydlexius**
## Summary

Emit a startup banner with the version and the effective (post-override) config/env at session start, with secrets redacted - so an operator can confirm what the process is actually running with.

- New `internal/config/redact.go` (single source of truth for sensitive field paths: `api.token`, `server.webhook_api_keys`) and `internal/config/render.go` (`FormatConfigText` console output + `ConfigToSlogAttrs` structured slog; both redact).
- Source tracking: each value is annotated `(env)` / `(cli)` when it came from an env var or a CLI override (`GetSetEnvVars` + `fieldEnvVars`).
- Banner wired into both `runFetch` and `runServe`, after `initLogging` and after CLI overrides, so it reflects the effective config.
- `internal/logging/redact.go`: added the plural `webhook_api_keys` to the slog redaction backstop.

UAT (live run, maintainer-verified): the Musixmatch token and webhook key render as `[REDACTED]` in both the human banner and the structured slog path; `--cooldown`/`--outdir` overrides show with `(cli)`, env values with `(env)`; version line emits.

## Linked issue

Closes #214

## Test plan

- [x] `make gate` GREEN; patch coverage 83.33%
- [x] TDD: redaction (token + webhook, both formats), render/source-tracking, banner emission
- [x] Hostile review (lead-driven): no secret leak in either format; both `runServe` callers updated; SHIP-READY
- [x] UAT: live banner verified (secrets redacted, source tags, version)
- [ ] Codoki auto-review

## Summary by CodeRabbit

* **New Features**
  * Added startup banner displaying application version and configuration upon launch
  * Configuration now shows source annotations indicating whether values came from CLI flags or environment variables

* **Improvements**
  * Enhanced logging with structured startup information display
  * Improved security by redacting sensitive values (API tokens, webhook keys) in logs and console output

* **Tests**
  * Added comprehensive test coverage for startup banner output and configuration rendering
  * Added tests validating sensitive data redaction

### CODE REVIEW

**Comment by sydlexius** [comment]
@coderabbitai review

**Comment by sydlexius** [comment]
@codoki The three High "Potential secret exposure" guardrail findings are false positives - no secret value is emitted at any of them:

- `internal/config/render.go:78` - this is the text-banner format line `webhook_api_keys = %s%s`. The value is rendered through `sliceVal`, which returns `[REDACTED]` for a non-empty sensitive field (or `[]` when empty). The only literal here is the config key *name*, not a secret.
- `internal/config/render.go:240` - the slog `sliceAttr` for `server.webhook_api_keys`. It redacts via `IsSensitiveConfigKey` and emits `[REDACTED]`; the actual keys are never logged in plaintext.
- `internal/config/render_test.go:30` - `cfg.API.Token = "supersecret"` is a test fixture in `TestFormatConfigText_RedactsToken`. The test asserts that the token does NOT appear in the rendered output, i.e. it verifies redaction. It is not a real credential and is the opposite of an exposure.

Redaction coverage for `api.token` and `server.webhook_api_keys` is exercised in both the text and structured-log paths. Please clear the strict-mode block on these three. The Medium empty-slice finding is valid and is acknowledged separately on its thread.

File: internal/config/render.go [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+	// floatAttr renders a float64 field with optional source annotation.
+	floatAttr := func(key, path string, val float64) slog.Attr {
+		if a := ann(path); a != "" {
+			return slog.String(key, fmt.Sprintf("%g%s", val, a))
+		}
+		return slog.Float64(key, val)
+	}
+
+	// sliceAttr renders a slice field, redacting sensitive values.
+	sliceAttr := func(key, path string, vals []string) slog.Attr {
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/config/render.go]: @codoki Accepted - this is a valid consistency gap. `sliceAttr` renders an empty slice as `""`, so an annotated empty slice shows a bare `(cli)`/`(env)`, while the text path renders `[]`. We will align `sliceAttr` to emit `[]` for empty (sensitive and non-sensitive) before appending the annotation, and add the `ConfigToSlogAttrs` test you suggested (empty -> `[]`, annotation appends to `[]`). Resolving this thread; the change is tracked for the next fix pass on this PR.

File: internal/config/config.go [hunk 2]
```diff
@@ -881,6 +881,75 @@ func clampMissBackoff(cfg *Config) {
...
+	for path, vars := range fieldEnvVars {
+		for _, v := range vars {
+			if os.Getenv(v) != "" {
+				result[path] = true
+				break
+			}
+		}
+	}
+	return result
+}
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/config/config.go]: @coderabbitai Fixed in 852ff75. `applyEnvOverrides` now records the set of config paths whose override **actually applied** (success branches only; rejected invalid values are not recorded). The new `LoadWithSources` threads that `appliedEnv` map out to `logStartupBanner`, which uses it as the env-source hint instead of `GetSetEnvVars` (now removed). A field whose env var is set but invalid (rejected) is therefore no longer annotated `(env)`. Added `TestLoadWithSources_EnvHintReflectsAppliedOverrides` covering valid-applies vs invalid-rejected.

File: internal/config/render.go [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+		return slog.Bool(key, val)
+	}
+
+	// floatAttr renders a float64 field with optional source annotation.
+	floatAttr := func(key, path string, val float64) slog.Attr {
+		if a := ann(path); a != "" {
+			return slog.String(key, fmt.Sprintf("%g%s", val, a))
+		}
+		return slog.Float64(key, val)
+	}
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/config/render.go]: @coderabbitai Fixed in 852ff75. `intAttr`/`boolAttr`/`floatAttr` now **always** emit the typed slog value (`slog.Int`/`slog.Bool`/`slog.Float64`) and, when a source applies, an additional sibling attr `<key>_source` carrying the bare `"cli"`/`"env"`. The log key type is stable across runs; the annotation no longer mutates the value type. String/slice/token fields stay inline-annotated (they are already strings and do not type-vary). Added `TestConfigToSlogAttrs_TypedValuesStableWithSourceSibling` asserting kind preservation plus sibling presence/absence/value.

File: internal/config/render.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/config/render.go]: @codoki This empty-slice rendering was fixed in F3: `sliceAttr` now renders empty slices as `[]` (both sensitive and non-sensitive) before appending the annotation, aligned with `FormatConfigText`. Covered by `TestConfigToSlogAttrs_EmptySliceRendersBrackets`. Resolving.

**Comment by sydlexius** [comment]
@coderabbitai approve
